packages/common/src/search/_internal/hubDiscussionsHelpers/processPostFilters.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ISearchPosts,
+  PostStatus,
+  PostType,
+  SharingAccess,
+} from "../../../discussions/api/types";
+import { IDateRange, IFilter } from "../../types";
+import { toEnum, toEnums } from "../hubEventsHelpers/toEnumConverters";
+import { flattenFilters } from "./processChannelFilters";
+
+/**
+ * Builds a Partial<ISearchPosts> given an Array of IFilter objects
+ * @param filters An Array of IFilter
+ * @returns A Partial<ISearchPosts> for the given Array of IFilter objects
+ */
+export function processPostFilters(filters: IFilter[]): Partial<ISearchPosts> {
+  const flattenedFilters = flattenFilters(filters);
+  const processedFilters: Partial<ISearchPosts> = {};
+
+  // access
+  if (flattenedFilters.access?.length) {
+    processedFilters.access = toEnums(
+      flattenedFilters.access as string[],
+      SharingAccess
+    );
+  }
+
+  // body
+  if (flattenedFilters.body?.length) {
+    processedFilters.body = flattenedFilters.body[0] as string;
+  }
+
+  // channels
+  if (flattenedFilters.channel?.length) {
+    processedFilters.channels = flattenedFilters.channel as string[];
+  }
+
+  // creator
+  if (flattenedFilters.owner?.length) {
+    processedFilters.creator = flattenedFilters.owner[0] as string;
+  }
+
+  // discussion
+  if (flattenedFilters.discussion?.length) {
+    processedFilters.discussion = flattenedFilters.discussion[0] as string;
+  }
+
+  // editor
+  if (flattenedFilters.editor?.length) {
+    processedFilters.editor = flattenedFilters.editor[0] as string;
+  }
+
+  // status
+  if (flattenedFilters.status?.length) {
+    processedFilters.status = toEnums(
+      flattenedFilters.status as string[],
+      PostStatus
+    );
+  }
+
+  // title
+  if (flattenedFilters.title?.length) {
+    processedFilters.title = flattenedFilters.title[0] as string;
+  }
+
+  // parents
+  if (flattenedFilters.parentId?.length) {
+    processedFilters.parents = flattenedFilters.parentId as string[];
```

Comment:
We have 3 scenarios we need to account for:

1. Searching for _both_ original posts (posts with no parent id) and replies. To do so, we'd pass `parentId: undefined` (or exclude `parentId` altogether)
2. Searching for replies to a specific original post. To do so, we'd pass `parentId: ['originalPostId']`
3. Searching for original posts _only_. To do this, we'd pass `parentId: []`

This code currently allows for scenarios 1 & 2 but not 3. Given that, I think we should simplify line 67 to just:

`if (flattenedFilters.parentId) {`

or

`if (Array.isArray(flattenedFilters.parentId)) {`